### PR TITLE
add subsection 'Best Practices when creating Pull Request'

### PR DESCRIPTION
**Pull request by shahzebsiddiqui**
This PR will create a subsection to document best practices when creating a pull request. I figure this would be helpful for new contributors when contributing changes.

@szuananwar can you please review my proposed changes

### CODE REVIEW

**Comment by szuananwar** [comment]
Sure!

**Comment by shahzebsiddiqui** [comment]
you can see the generated docs see https://buildtest--1531.org.readthedocs.build/en/1531/contributing/code_contribution_guide.html#best-practices-when-creating-pull-request

File: docs/contributing/code_contribution_guide.rst [hunk 1]
```diff
@@ -90,6 +90,19 @@ branch which is your feature branch pushed at your fork.
 .. note::
     Do not push to ``master`` or ``devel`` branch on your fork or upstream.
 
+Best Practices when creating Pull Request
```
Comment (szuananwar) [inline docs/contributing/code_contribution_guide.rst]: Please capitalize the "W" in When and "C" in creating.

File: docs/contributing/code_contribution_guide.rst [hunk 2]
```diff
@@ -90,6 +90,19 @@ branch which is your feature branch pushed at your fork.
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- It's good practice to link PR to an issue during commit message. Such as stating ``Fix #132`` for fixing issue 132.
+
+- Please create a meaningful title and PR description to help outline your proposed changes.
+
+- Assign PR to yourself when creating the issue. You should @ mention (`@shahzebsiddiqui <https://github.com/shahzebsiddiqui>`_) the project maintainers to get their attention.
+
+- If your PR is not ready for review, please add ``WIP:`` to your PR title to indicate it's a work in progress and make it a draft PR. This will prevent maintainers from reviewing your PR until it's ready.
```
Comment (szuananwar) [inline docs/contributing/code_contribution_guide.rst]: Line 102: please  add "that" after "to indicate"

File: docs/contributing/code_contribution_guide.rst [hunk 3]
```diff
@@ -90,6 +90,19 @@ branch which is your feature branch pushed at your fork.
...
+
+- Check the CI checks corresponding to your PR to ensure all checks are passed. If you see any failures, please fix them especially regression test failures.
```
Comment (szuananwar) [inline docs/contributing/code_contribution_guide.rst:104]: Please add comma after especially.

File: docs/contributing/code_contribution_guide.rst [hunk 4]
```diff
@@ -143,25 +156,15 @@ Once you have synced your branch push your changes and check if file conflicts a
 General Tips
 -------------
 
-1. It's good practice to link PR to an issue during commit message. Such as
-stating ``Fix #132`` for fixing issue 132.
-
-2. If you have an issue, ask your question in slack before reporting issue. If
-your issue is not resolved check any open issues for resolution before creating
-a new issue.
+- If you have an issue, ask your question in slack before reporting issue. If your issue is not resolved check any open issues for resolution before creating a new issue.
```
Comment (szuananwar) [inline docs/contributing/code_contribution_guide.rst]: Please add "the" after "before reporting"

File: docs/contributing/code_contribution_guide.rst [hunk 5]
```diff
@@ -143,25 +156,15 @@ Once you have synced your branch push your changes and check if file conflicts a
...
 
-3. For new features or significant code refactor please notify maintainers and
-open an issue before working on task to keep everyone informed.
+- For new features or significant code refactor please notify maintainers and open an issue before working on task to keep everyone informed.
```
Comment (szuananwar) [inline docs/contributing/code_contribution_guide.rst]: Please pluralize refractor and task and add a comma after refractors.

File: docs/contributing/code_contribution_guide.rst [hunk 6]
```diff
@@ -143,25 +156,15 @@ Once you have synced your branch push your changes and check if file conflicts a
...
 
-4. If you open an issue, please respond back during discussion, if there is no
-activity the issue will be closed.
+- If you open an issue, please respond back during discussion, if there is no activity the issue will be closed.
```
Comment (szuananwar) [inline docs/contributing/code_contribution_guide.rst]: Please add "the" after during.

File: docs/contributing/code_contribution_guide.rst [hunk 7]
```diff
@@ -143,25 +156,15 @@ Once you have synced your branch push your changes and check if file conflicts a
...
 
-5. Please refrain from opening duplicate issue, check if there is an existing
-issue addressing similar problem, instead you can participate in discussion in
-the issue or contact appropriate individuals directly in slack.
+- Please refrain from opening duplicate issue, check if there is an existing issue addressing similar problem. You can ask questions in slack to report your issue or contact project maintainers.
```
Comment (szuananwar) [inline docs/contributing/code_contribution_guide.rst]: Please add "a" after opening. Pluralize problem.

File: docs/contributing/code_contribution_guide.rst (hunk 1 above)
Comment (shahzebsiddiqui) [reply docs/contributing/code_contribution_guide.rst]: fixed

File: docs/contributing/code_contribution_guide.rst (hunk 2 above)
Comment (shahzebsiddiqui) [reply docs/contributing/code_contribution_guide.rst]: fixed

File: docs/contributing/code_contribution_guide.rst (hunk 4 above)
Comment (shahzebsiddiqui) [reply docs/contributing/code_contribution_guide.rst]: fixed

File: docs/contributing/code_contribution_guide.rst (hunk 6 above)
Comment (shahzebsiddiqui) [reply docs/contributing/code_contribution_guide.rst]: fixed

File: docs/contributing/code_contribution_guide.rst (hunk 7 above)
Comment (shahzebsiddiqui) [reply docs/contributing/code_contribution_guide.rst]: fixed

Comment (szuananwar) [reply docs/contributing/code_contribution_guide.rst]: Please add "a" after opening and not before it.

Comment (shahzebsiddiqui) [reply docs/contributing/code_contribution_guide.rst]: fixed

**Comment by shahzebsiddiqui** [comment]
@szuananwar please see now, i addressed your feedback. Thanks for your feedback

**Comment by szuananwar** [comment]
Thanks for making all changes, it look good.

File: docs/contributing/code_contribution_guide.rst [hunk 8]
```diff
@@ -143,25 +156,15 @@ Once you have synced your branch push your changes and check if file conflicts a
...
+- For new features or significant code refactors, please notify maintainers and open an issue before working on task to keep everyone informed.
 
-4. If you open an issue, please respond back during discussion, if there is no
-activity the issue will be closed.
+- If you open an issue, please respond back during the discussion, if there is no activity the issue will be closed.
 
-5. Please refrain from opening duplicate issue, check if there is an existing
-issue addressing similar problem, instead you can participate in discussion in
-the issue or contact appropriate individuals directly in slack.
+- Please refrain from opening a duplicate issue, check if there is an existing issue addressing similar problems. You can ask questions in slack to report your issue or contact project maintainers.
```
Comment (szuananwar) [inline docs/contributing/code_contribution_guide.rst:165]: look good!

**Review by szuananwar — APPROVED** [review]

**Pull request merged**